glommio/src/executor/mod.rs

```diff
@@ -1110,11 +1111,21 @@ impl LocalExecutor {
         self.reactor.need_preempt()
     }
 
+    pub(crate) fn need_preempt_check_timer(&self) -> bool {
```

Comment:
I am not a fan of this, but I have to think a bit more about the best way to tackle it.

Is the reason you need it the fact that by consuming the cqe in the latency ring, need preempt is now zero ?

One of the reasons I don't like it, besides the special casing, is that `need_preempt()` is supposed to be about more than timers. But rather, any event in the latency ring.

Also the fact that this works "only between task queues" is an artifact of the current implementation. Up until very recently rush_dispatch could have been called inside a task queue (thirstycrow temporarily removed it)

Reply:
> Is the reason you need it the fact that by consuming the cqe in the latency ring, need preempt is now zero ?

Yes. `need_preempt` works under the assumption that a timer is always scheduled on the latency ring on the way out of the reactor. If it isn't there (because we polled the ring from a task, for instance) then the preemption mechanism breaks. 

I added this variant to be more robust and cover that use case.

> Also the fact that this works "only between task queues"

It works from everywhere. I use it between queues because that's enough in practice.

>  Up until very recently rush_dispatch could have been called inside a task queue (thirstycrow temporarily removed it)

Are you sure? From what I can see `rush_dispatch` is being used in the net code